pkg/pillar/cmd/zedrouter/pubsubhandlers.go

```diff
@@ -629,3 +629,134 @@ func (z *zedrouter) handleAppInstDelete(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	z.log.Functionf("handlePatchEnvelopeDelete: (UUID: %s)", key)
+
+	z.handlePatchEnvelopeImpl(types.PatchEnvelopeInfoList{})
+
+	z.log.Functionf("handlePatchEnvelopeDelete(%s) done", key)
+}
+
+func (z *zedrouter) handleContentTreeStatusCreate(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	contentTree := statusArg.(types.ContentTreeStatus)
+	z.log.Functionf("handleContentTreeStatusCreate: (UUID: %s, name:%s)",
+		key, contentTree.DisplayName)
+
+	z.patchEnvelopes.UpdateContentTree(contentTree, false)
+
+	go func() {
+		z.patchEnvelopes.UpdateStateNotificationCh() <- struct{}{}
+	}()
+
+	z.log.Functionf("handleContentTreeStatusCreate(%s) done", key)
+}
+
+func (z *zedrouter) handleContentTreeStatusModify(ctxArg interface{}, key string,
+	statusArg interface{}, oldStatusArg interface{}) {
+
+	contentTree := statusArg.(types.ContentTreeStatus)
+	z.log.Functionf("handleContentTreeStatusModify: (UUID: %s), name:%s",
+		key, contentTree.DisplayName)
+
+	z.patchEnvelopes.UpdateContentTree(contentTree, false)
+
+	go func() {
+		z.patchEnvelopes.UpdateStateNotificationCh() <- struct{}{}
+	}()
+
+	z.log.Functionf("handleContentTreeStatusModify(%s) done", key)
+
+}
+
+func (z *zedrouter) handleContentTreeStatusDelete(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	contentTree := statusArg.(types.ContentTreeStatus)
+	z.log.Functionf("handleVolumeStatusDelete: (UUID: %s, name:%s)",
+		key, contentTree.DisplayName)
+
+	z.patchEnvelopes.UpdateContentTree(contentTree, true)
+
+	go func() {
+		z.patchEnvelopes.UpdateStateNotificationCh() <- struct{}{}
+	}()
+
+	z.log.Functionf("handleContentTreeStatusDelete(%s) done", key)
+
+}
+
+func (z *zedrouter) handleVolumeStatusCreate(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	volume := statusArg.(types.VolumeStatus)
+	z.log.Functionf("handleVolumeStatusCreate: (UUID: %s, name:%s)",
+		key, volume.DisplayName)
+
+	z.patchEnvelopes.UpdateVolumeStatus(volume, false)
+
+	go func() {
+		z.patchEnvelopes.UpdateStateNotificationCh() <- struct{}{}
+	}()
+
+	z.log.Functionf("handleVolumeStatusCreate(%s) done", key)
+}
+
+func (z *zedrouter) handleVolumeStatusModify(ctxArg interface{}, key string,
+	statusArg interface{}, oldStatusArg interface{}) {
+
+	volume := statusArg.(types.VolumeStatus)
+	z.log.Functionf("handleVolumeStatusModify: (UUID: %s), name:%s",
+		key, volume.DisplayName)
+
+	z.patchEnvelopes.UpdateVolumeStatus(volume, false)
+
+	go func() {
+		z.patchEnvelopes.UpdateStateNotificationCh() <- struct{}{}
+	}()
+
+	z.log.Functionf("handleVolumeStatusModify(%s) done", key)
+}
+
+func (z *zedrouter) handleVolumeStatusDelete(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	volume := statusArg.(types.VolumeStatus)
+	z.log.Functionf("handleVolumeStatusDelete: (UUID: %s, name:%s)",
+		key, volume.DisplayName)
+
+	z.patchEnvelopes.UpdateVolumeStatus(volume, true)
+
+	go func() {
```

Comment:
No there were a few more. 
Probably makes sense to define some triggerPEUpdate function which does a conditional send.

Reply:
defined triggerPEUpdate function and used it instead of spawning goroutines :)